python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -0,0 +1,241 @@
+from lsst.daf.base import PropertyList
+
+
+@dataclass
+class FocalPlaneRegionExtendedPsf:
+    """Single extended PSF over a focal plane region.
+
+    The focal plane region is defined through a list
+    of detectors.
+
+    Parameters
+    ----------
+    extended_psf_image : `lsst.afw.image.MaskedImageF`
+        Image of the extended PSF model.
+    detector_list : `list` [`int`]
+        List of detector IDs that define the focal plane region over which this
+        extended PSF model has been built (and can be used).
+    """
+    extended_psf_image: afwImage.MaskedImageF
+    detector_list: List[int]
+
+
+class ExtendedPsf:
+    """Extended PSF model.
+
+    Each instance may contain a default extended PSF, a set of extended PSFs
+    that correspond to different focal plane regions, or both. At this time,
+    focal plane regions are always defined as a subset of detectors.
+
+    Parameters
+    ----------
+    default_extended_psf : `lsst.afw.image.MaskedImageF`
+        Extended PSF model to be used as default (or only) extended PSF model.
+    """
+    def __init__(self, default_extended_psf=None):
+        self.default_extended_psf = default_extended_psf
+        self.focal_plane_regions = {}
+        self.detectors_focal_plane_regions = {}
+
+    def add_regional_extended_psf(self, extended_psf_image, region_name, detector_list):
+        """Add a new focal plane region, along wit hits extended PSF, to the
+        ExtendedPsf instance.
+
+        Parameters
+        ----------
+        extended_psf_image : `lsst.afw.image.MaskedImageF`
+            Extended PSF model for the region.
+        region_name : `str`
+            Name of the focal plane region. Will be converted to all-uppercase.
+        detector_list : `list` [`int`]
+            List of IDs for the detectors that define the focal plane region.
+        """
+        region_name = region_name.upper()
+        if region_name in self.focal_plane_regions:
+            raise ValueError(f"Region name {region_name} is already used by this ExtendedPsf instance.")
+        self.focal_plane_regions[region_name] = FocalPlaneRegionExtendedPsf(
+            extended_psf_image=extended_psf_image, detector_list=detector_list)
+        for det in detector_list:
+            self.detectors_focal_plane_regions[det] = region_name
+        return None
+
+    def __call__(self, detector=None):
+        """Return the appropriate extended PSF.
+
+        If the instance contains no extended PSF defined over focal plane
+        regions, the default extended PSF will be returned regardless of
+        whether a detector ID was passed as argument.
+
+        Parameters
+        ----------
+        detector : `int`, optional
+            Detector ID. If this instance contains extended PSFs defined over
+            focal plane regions, the extended PSF model for the region that
+            contains ``detector`` is returned. If not, the default extended PSF
+            is returned.
```

Comment:
You should have a Returns section for non-trivial (non None) returns if you have to explain the returned object in any detail. https://developer.lsst.io/python/numpydoc.html?highlight=returns#py-docstring-returns